### PR TITLE
Upgrade

### PR DESCRIPTION
**Pull request by iky**
* A few formatting improvements
  * Apply [black](https://github.com/ambv/black)
  * trim trailing white spaces
  * sort imports
  * new PEP8s
* Add support for Python 3.7 and remove support for Python 3.4
* Make tests working with both major Nameko versions - https://github.com/iky/nameko-slack/commit/d34f628639b6e7fdbf53fe0c9f610e7baf2b7c43

### CODE REVIEW

**Review by mattbennett — COMMENTED** [review]
Might want to consider adding a `.pre-commit.yaml` config file and/or enforcing black in CI.

Check out nameko/nameko-grpc for an example project set up that way.

File: .travis.yml [hunk 1]
```diff
@@ -1,27 +1,44 @@
...
+  - python: 3.6
+    env: NAMEKO_VERSION===2.10.0
+  - python: 3.6
+    env: NAMEKO_VERSION===2.9.1
+  - python: 3.6
+    env: NAMEKO_VERSION===2.9.0
+  - python: 3.6
+    env: NAMEKO_VERSION===2.8.5
+  - python: 3.6
+    env: NAMEKO_VERSION="==2.8.1 pytest<3.3.0"
```
Comment (mattbennett) [inline .travis.yml]: `DEPS=`? ;)

**Review by mattbennett — APPROVED** [review]

**Comment by mattbennett** [comment]
One other thought related to this --

It might be nice to have a general "prerelease" item in the matrix, so you are always running against the latest prerelease version, rather than hardcoding to a specific version.

With a general "prerelease" item you could also add a cron-based execution to Travis, which would alert early if a prerelease version of Nameko introduces incompatibilities.

File: .travis.yml (hunk 1 above)
Comment (iky) [reply .travis.yml]: :+1:  good suggestion .)

**Review by mattbennett — APPROVED** [review]

**Comment by iky** [comment]
@mattbennett I replaced the hardcoded prerealase with generic prerelease installation. I also added a cronjob running daily but I marked the prerelease test as allow-failure so it does not fail builds with tests passing against normal nameko releases.

**Comment by iky** [comment]
I changed the cronjob to weekly .)

**Pull request merged**